apps/minifront-v2/shared/hooks/use-connection.ts

```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect } from 'react';
+import { penumbra } from '@shared/lib/penumbra';
+import { PenumbraClient } from '@penumbra-zone/client';
+
+export const useIsConnected = (): boolean => {
+  const [connected, setConnected] = useState(false);
+
+  useEffect(() => {
+    // Check initial connection state
+    setConnected(Boolean(penumbra.connected));
+
+    // Listen for connection state changes
+    const unsubscribe = penumbra.onConnectionStateChange(() => {
+      setConnected(Boolean(penumbra.connected));
+    });
+
+    return unsubscribe;
+  }, []);
+
+  return connected;
+};
+
+export const useConnectWallet = () => {
+  const connectWallet = async () => {
+    console.log('Connect wallet clicked');
```

Comment:
cleanup: while the linter is not fully set up for minifront v2, it's best to keep the logs out of the final PR